### PR TITLE
Feature rec val validators

### PR DESCRIPTION
**Pull request by darh**
Please take a look (all but generated event/*go) files

### CODE REVIEW

File: compose/service/record.go [hunk 1]
```diff
@@ -326,6 +335,43 @@ func (svc record) Export(filter types.RecordFilter, enc Encoder) error {
 }
 
 func (svc record) Create(new *types.Record) (rec *types.Record, err error) {
+	var invokerID = auth.GetIdentityFromContext(svc.ctx).Identity()
+
+	// Runs value sanitization, sets values that should used
```
Comment (tjerman) [inline compose/service/record.go]: `should used` -> `should be used` :')

File: compose/service/values/sanitizer.go [hunk 2]
```diff
@@ -86,42 +71,42 @@ func (s sanitizer) Run(m *types.Module, vv types.RecordValueSet) (out types.Reco
...
+			//	v = s.sRecord(v, f, m)
+			//case "select":
+			//	v = s.sSelect(v, f, m)
+			//case "string":
+			//	v = s.sString(v, f, m)
+			//case "url":
+			//	v = s.sUrl(v, f, m)
+			//case "user":
+			//	v = s.sUser(v, f, m)
 		}
```
Comment (tjerman) [inline compose/service/values/sanitizer.go]: What's up with all of these?

File: compose/service/values/shared.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	"regexp"
+	"time"
+)
+
+const (
+	strBoolTrue  = "1"
+	strBoolFalse = "0"
+
+	datetimeInputFormatDate = "2006-01-02"
+	datetimeInputFormatTime = "15:04:05"
```
Comment (tjerman) [inline compose/service/values/shared.go]: should we not handle TZ?

File: compose/service/values/shared.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	fieldOpt_Datetime_onlyPastValues   = "onlyPastValues"
+
+	fieldOpt_Number_precision = "precision"
+
+	fieldOpt_Url_onlySecure = "onlySecure"
+)
+
+var (
+	// value resembles something that can be true
+	truthy = regexp.MustCompile(`^(t(rue)?|y(es)?|1)$`)
```
Comment (tjerman) [inline compose/service/values/shared.go]: add `i` (case insensitive) flag for good measure

File: compose/service/values/sanitizer.go (hunk 2 above)
Comment (darh) [reply compose/service/values/sanitizer.go]: placeholders.

File: compose/service/record.go (hunk 1 above)
Comment (darh) [reply compose/service/record.go]: ty.

File: compose/service/values/shared.go (hunk 3 above)
Comment (darh) [reply compose/service/values/shared.go]: we do (in sanitizer), but these are internal datetime formats

File: compose/service/values/shared.go (hunk 4 above)
Comment (darh) [reply compose/service/values/shared.go]: we do  it on the value.

**Pull request merged**